gitgud/operations.py

```diff
@@ -159,51 +159,53 @@ def commit(self, commit_message, parents, time_offset):
 
     def file_in_commit(self, commit, filepath):
         commit = self.repo.commit(commit)
+        if not isinstance(filepath, str):
+            filepath = str(filepath)
         return filepath in commit.tree
 
-    def get_commit_content(self, commit, filepath):
-        if commit in self._streamed_content:
-            return self._streamed_content[commit]
-
+    def get_commit_file_content(self, commit, filepath):
         commit = self.repo.commit(commit)
-
         if not isinstance(filepath, str):
             filepath = str(filepath)
 
-        commit_content = (commit.tree / filepath).data_stream.read().decode("ascii")  # noqa: E501
-        self._streamed_content[commit] = commit_content
-        return commit_content
-
-    def get_staging_area(self):
-        return self.get_tracked_changes("staging")
+        if commit in self._streamed_content and \
+                filepath in self._streamed_content[commit]:
+            return self._streamed_content[commit][filepath]
 
-    def get_working_area(self):
-        changes = self.get_tracked_changes("working")
-        changes["added"] += self.repo.untracked_files
-        return changes
+        commit_content = (commit.tree / filepath) \
+            .data_stream.read().decode("ascii")
+        if commit not in self._streamed_content:
+            self._streamed_content[commit] = {}
 
-    def get_tracked_changes(self, change_type):
-        if change_type == "staging":
-            staged_files = self.repo.index.diff("HEAD")
-        elif change_type == "working":
-            staged_files = self.repo.index.diff(None)
-        else:
-            raise KeyError
+        self._streamed_content[commit][filepath] = commit_content
+        return commit_content
 
-        change_key_map = {
-            "added": "A",
-            "deleted": "D",
-            "modified": "M",
-            "renamed": "R"
-        }
+    def get_commit_content(self, commit):
+        content = {}
+        for filepath in commit.tree:
+            content.update(
+                {filepath: self.get_commit_file_content(commit, filepath)}
+            )
+        return content
+
+    def get_staging_content(self):
+        content = {}
+        for (path, _stage), entry in self.repo.index.entries.items():
```

Comment:
```
>>> list(op.repo.index.iter_blobs())[0]
(0, <git.Blob "788df9fc65e5d61b14b96c96fa9a0e2cd241b109">)
>>> list(op.repo.index.iter_blobs())[0][1]
<git.Blob "788df9fc65e5d61b14b96c96fa9a0e2cd241b109">
>>> list(op.repo.index.iter_blobs())[0][1].path
'1.txt'
>>> list(op.repo.index.iter_blobs())[0][1].data_stream.read()
b"Hello, I'm an auto-generated file!"
```

Reply:
We want to use `iter_blobs` instead. The first item in that tuple is the stage, as defined in the [3-way merge documentation ](https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-read-tree.html)